src/Components/Components/src/ParameterCollectionExtensions.cs

```diff
@@ -120,8 +208,34 @@ public WritersForType(Type targetType)
                     }
 
                     WritersByName.Add(propertyName, propertySetter);
+
+                    if (parameterAttribute != null && parameterAttribute.CaptureUnmatchedValues)
+                    {
+                        // This is an "Extra" parameter.
+                        //
+                        // There should only be one of these.
+                        if (CaptureUnmatchedValuesWriter != null)
+                        {
+                            ThrowForMultipleCaptureUnmatchedValuesParameters(targetType);
+                        }
+
+                        // It must be able to hold a Dictionary<string, object> since that's what we create.
+                        if (!propertyInfo.PropertyType.IsAssignableFrom(typeof(Dictionary<string, object>)))
```

Comment:
Is the condition inverted here or am I reading this incorrectly:

` if (!typeof(Dictionary<string, object>).IsAssignableFrom(propertyInfo.PropertyType))`

Reply:
It's correct and it has tests that are passing. We don't need to assign the property type **into** a Dictionary<>, we need to assign a Dictionary<> into the property type.